server/sonar-server/src/main/java/org/sonar/server/permission/PermissionService.java

```diff
@@ -74,11 +76,25 @@ public void applyDefaultPermissionTemplate(DbSession session, String componentKe
     } else {
       userSession.checkPermission(GlobalPermissions.PROVISIONING);
     }
-    permissionRepository.applyDefaultPermissionTemplate(session, component);
+
+    Integer currentUserId = userSession.getUserId();
+    Long userId = Qualifiers.PROJECT.equals(component.qualifier()) && currentUserId != null ? currentUserId.longValue() : null;
+    permissionRepository.applyDefaultPermissionTemplate(session, component, userId);
     session.commit();
     indexProjectPermissions();
   }
 
+  public boolean wouldCurrentUserHavePermissionWithDefaultTemplate(DbSession dbSession, String permission, @Nullable String branch, String projectKey, String qualifier) {
+    if (userSession.hasPermission(permission)) {
+      return true;
+    }
+
+    String effectiveKey = ComponentKeys.createKey(projectKey, branch);
+
+    Long userId = userSession.getUserId() == null ? null : userSession.getUserId().longValue();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'getUserId' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)